Tests.ReactView/InnerViewTests.cs

```diff
@@ -15,18 +16,72 @@ public class InnerViewTests : ReactViewTestBase {
         [Test(Description = "Tests inner view load")]
         public async Task InnerViewIsLoaded() {
             await Run(async () => {
-                var taskCompletionSource = new TaskCompletionSource<bool>();
-                
+                var taskCompletionSource = new TaskCompletionSource<bool>();
+
                 TargetView.InnerView.Loaded += () => taskCompletionSource.TrySetResult(true);
 #if DEBUG
-                TargetView.Ready += () => TargetView.InnerView.Load();
-#endif
+                TargetView.Ready += () => TargetView.InnerView.Load();
+#endif
                 TargetView.InnerView.Load();
+
                 var isLoaded = await taskCompletionSource.Task;
 
                 Assert.IsTrue(isLoaded, "Inner view module was not loaded!");
             });
-        }
+        }
+
+        [Test(Description = "Tests inner view reload when web view is reloaded")]
+        public async Task InnerViewIsReloadedWhenWebviewIsReloaded() {
+            await Run(async () => {
+                // Load view for the first time
+                var taskCompletionSource = new TaskCompletionSource<bool>();
+                TargetView.InnerView.Loaded += () => {
+                    taskCompletionSource.TrySetResult(true);
+                };
+
+                TargetView.InnerView.Load();
+                await taskCompletionSource.Task;
+
+                // Reload view
+                taskCompletionSource = new TaskCompletionSource<bool>();
+                TargetView.ExecuteMethod("reload");
+                var isReloaded = await taskCompletionSource.Task;
+
+                Assert.IsTrue(isReloaded, "Inner view module was not reloaded!");
+            });
+        }
+
+        [Test(Description = "Tests inner view is collected when destroyed")]
+        public async Task InnerViewIsCollectedWhenDestroyed() {
+            await Run(async () => {
+                var weakRef = new WeakReference(TargetView.InnerView);
+
+                // Load view for the first time
+                var taskCompletionSource = new TaskCompletionSource<bool>();
+                TargetView.InnerView.Loaded += () => {
+                    taskCompletionSource.TrySetResult(true);
+                };
+
+                TargetView.InnerView.Load();
+                await taskCompletionSource.Task;
+
+                // Reload view
+                var targetViewCompletionSource = new TaskCompletionSource<bool>();
+                TargetView.ExecuteMethod("setShouldRenderInnerView", false);

```

Comment:
TargetView.ExecuteMethod("setShouldRenderInnerView", false); this does not make sense... the state will go away with reload

Reply:
true... but I can't make it work by simply changing the props... :/

Reply:
i'm not confident that this is really working...